sympy/utilities/tests/test_wester.py

```diff
@@ -2181,21 +2182,25 @@ def test_U4():
     assert diff(x**n, x, n).rewrite(factorial) == factorial(n)
 
 
-@XFAIL
 def test_U5():
-    # https://github.com/sympy/sympy/issues/6681
-    # f(g(x)).diff(x,2) returns Derivative(g(x), x)**2*Subs(Derivative(
-    #  f(_xi_1), _xi_1, _xi_1), (_xi_1,), (g(x),)) + Derivative(g(x), x, x)*
-    #  Subs(Derivative(f(_xi_1), _xi_1), (_xi_1,), (g(x),))
-    raise NotImplementedError("f(g(t)).diff(t,2) Subs not performed")
-
-
-@XFAIL
-def test_U6():
+    # issue 6681
+    t = symbols('t')
+    ans = (
+        Derivative(f(g(t)), g(t))*Derivative(g(t), (t, 2)) +
+        Derivative(g(t), t)*Derivative(f(g(t)), t, g(t)))
+    assert f(g(t)).diff(t,2) == ans
+    doans = (
+        Derivative(f(g(t)), g(t))*Derivative(g(t), (t, 2)) +
+        Derivative(g(t), t)**2*Derivative(f(g(t)), g(t), 2))
+    assert ans.doit() == doans
+
+
+def test_U6():  # XXX check
     h = Function('h')
-    # raises ValueError: Invalid limits given: (y, h(x), g(x))
     T = integrate(f(y), y, h(x), g(x))
-    T.diff(x)
+    assert T.diff(x) == (
```

Comment:
I think this should be
```
f(g(x))*g(x).diff(x)-f(h(x))*h(x).diff(x)
```

Reply:
That seems right to me.

Reply:
The previous or the new? If you look at the `Integral._eval_derivative` it is more complicated...but it for sure did not get it right as it is because `y` is still in the answer that was given for the U6 result. I have to think about how this is modified with multiple integrations.

Reply:
Sorry, I should have said that I agree. What you propose looks right to me.